crates/hyperswitch_domain_models/src/merchant_account.rs

```diff
@@ -350,6 +355,54 @@ impl From<MerchantAccountUpdate> for MerchantAccountUpdateInternal {
     }
 }
 
+#[cfg(all(feature = "v2", feature = "merchant_account_v2"))]
+impl From<MerchantAccountUpdate> for MerchantAccountUpdateInternal {
+    fn from(merchant_account_update: MerchantAccountUpdate) -> Self {
+        let now = date_time::now();
+
+        match merchant_account_update {
+            MerchantAccountUpdate::Update {
+                merchant_name,
+                merchant_details,
+                routing_algorithm,
+                publishable_key,
+                metadata,
+                frm_routing_algorithm,
+                payout_routing_algorithm,
+            } => Self {
+                merchant_name: merchant_name.map(Encryption::from),
+                merchant_details: merchant_details.map(Encryption::from),
+                frm_routing_algorithm,
+                routing_algorithm,
+                publishable_key,
+                metadata,
+                modified_at: Some(now),
```

Comment:
Can we make `modified_at` a required field in `MerchantAccountUpdateInternal` and remove the `Default` derive on `MerchantAccountUpdateInternal`?